stl/inc/vector

```diff
@@ -440,6 +440,183 @@ constexpr _Ty* _Unfancy_maybe_null(_Ty* _Ptr) noexcept { // do nothing for plain
     return _Ptr;
 }
 
+#ifdef __SANITIZE_ADDRESS__
+extern "C" void __sanitizer_annotate_contiguous_container(const void*, const void*, const void*, const void*);
+
+template <class _Vec, class = void>
+constexpr bool _Has_minimum_allocation_alignment = disjunction_v<bool_constant<alignof(typename _Vec::value_type) >= 8>,
+    is_same<typename _Vec::allocator_type, allocator<typename _Vec::value_type>>>;
+
+template <class _Vec>
+constexpr bool
+    _Has_minimum_allocation_alignment<_Vec, void_t<decltype(_Vec::allocator_type::_Minimum_allocation_alignment)>> =
+        _Vec::allocator_type::_Minimum_allocation_alignment >= 8;
+
+template <class _Ty, bool = _Has_minimum_allocation_alignment<_Ty>>
+class _NODISCARD _Asan_annotater {
+private:
+    using _Alloc      = typename _Ty::allocator_type;
+    using _Pointer    = typename _Ty::pointer;
+    using _Size_type  = typename _Ty::size_type;
+    using _Value_type = typename _Ty::value_type;
+
+public:
+    explicit constexpr _Asan_annotater(const _Ty& _Target) noexcept : _Target_vec(_Target), _Old_last(nullptr) {}
+
+    _CONSTEXPR20 _Asan_annotater(const _Ty& _Target, const _Size_type _New_size)

```

Comment:
As Casey suggests `__sanitizer_annotate_contiguous_container` is `noexcept`, then this is also `noexcept`

Reply:
[allocator.requirements.general]/7 tells us that comparisons/swaps/copies/and moves of fancy pointers must not throw, and [nullablepointer.requirements]/1 that the Cpp17NullablePointer (which requirements allocator pointers must also satisfy) operations must not throw, but notably neither of these cover addition or subtraction. That might be a defect in the Standard, but in any case the current state of the wording seems to allow `_Myend - Myfirst` to throw exceptions.